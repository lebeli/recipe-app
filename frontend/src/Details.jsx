import React, { Component } from "react";
import BackButton from "./BackButton";
import Ingredients from "./Ingredients";
import Instructions from "./Instructions";
import "./Details.scss";

class Details extends Component {
  constructor(props) {
    super(props);
    this.state = {
      pageNumber: 1
    };
  }

  render() {
    return (
      <div className="Details">
        <BackButton handleGoBack={this.props.handleGoBack} />
        <Ingredients />
        <Instructions />
      </div>
    );
  }
}

export default Details;
