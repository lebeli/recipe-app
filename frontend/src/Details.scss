@import "./styles/styles.scss";

.Details {

  .TeaserImage {
    height: 50vh;
    overflow: hidden;
    display: flex;
    justify-content: center;

    Img {
      height: 50vh;
      width: 100%;
      object-fit: cover;
    }
  }

  .DetailedInfo {
    padding: 5%;
    width: calc(100vw - 10%);
    display: flex;

    .TextInfo {
      width: 50%;
      bottom: 0;
      left: 0;
      flex: 50%;

      .Header {
        background: rgba(255, 255, 255, 0.5);
        height: 10%;
        justify-content: center;
        align-items: center;
        padding: 10px;
        display: flex;

        .HeaderContent {
          margin: 5px;
          width: 100%;
          text-align: center;
          display: flex;
          align-items: center;
          justify-content: center;
        }
      }

      Button {
        color: $primaryColor;
        padding-left: 0;
      }

      .Ingredients {
        h4 {
          margin: 0;
        }
        ul {
          list-style-type: none;
          padding: 0;
          margin: 0;
        }
      }
    }

    .FullImage {
      flex: 50%;
      display: flex;
      align-items: center;

      Img {
        width: 100%;
        object-fit: cover;
      }
    }
  }

  @media screen and (max-width: 450px) {
    .DetailedInfo {
      display: block;

      .TextInfo {
        width: 100%;
      }

      .FullImage {
        Img {
          width: 100%;
        }
      }
    }
  }
}
