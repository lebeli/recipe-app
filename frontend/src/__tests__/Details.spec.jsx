import React from "react";
import { shallow, mount } from "enzyme";
import renderer from "react-test-renderer";
import Details from "../Details";

describe("Recommender", () => {
  test("should render elementes", () => {
    const wrapper = mount(<Details />);
    expect(wrapper.find(".RecommenderImage"));
    expect(wrapper.find(".BackButton"));
    expect(wrapper.find(".RecommenderHeader"));
    expect(wrapper.find(".Ingredients"));
    expect(wrapper.find(".Instructions"));
  });

  test("renders correctly", () => {
    const tree = renderer.create(<Details />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
