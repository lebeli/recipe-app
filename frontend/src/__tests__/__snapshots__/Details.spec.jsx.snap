// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details renders correctly 1`] = `
<div
  className="Details"
>
  <div
    className="TeaserImage"
  >
    <img
      alt="finished dish"
    />
  </div>
  <div
    className="DetailedInfo"
  >
    <div
      className="TextInfo"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
          Zur√ºck
        </span>
      </button>
      <div
        className="MuiGrid-root MuiGrid-container"
        id="details_header"
      >
        <div
          className="MuiGrid-root MuiGrid-item"
          id="details_recipe_name"
        />
        <div
          className="MuiGrid-root MuiGrid-item"
          id="details_duration"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
            <path
              d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
            />
          </svg>
        </div>
      </div>
      <div
        className="Ingredients"
      >
        <h4>
          ZUTATEN
        </h4>
        <div>
          <span>
            300g
          </span>
          flour
        </div>
        <div>
          <span>
            1
          </span>
          cucumber
        </div>
      </div>
      <div
        className="Instructions"
      >
        <ol>
          <li>
            Banannaaaaa
          </li>
          <li>
            Gruuuu!
          </li>
          <li>
            Halloosn viawhbfiaw hfeiuwbfhbwe diqwbdhbvwei lfhea iugf46griwa zdegksg fuishfiu gfzsgdfjz sfv,jhbs&lt; fhksbc. kjsbdvjk bydskjvf sudhv ousgczs h!!
          </li>
        </ol>
      </div>
    </div>
    <div
      className="FullImage"
    >
      <img
        alt="Rezept Detail Bild - Freunde beim Essen"
        src="test-file-stub"
      />
    </div>
  </div>
</div>
`;
